app/components/tag_list_input_component/tag_list_input_controller.js

```diff
@@ -0,0 +1,65 @@
+import { Controller } from "stimulus";
+
+export default class extends Controller {
+  static targets = ["tagList", "newTag", "template", "list"];
+
+  addTag(event) {
+    // prevent hotkey form submitting the form (default action for "enter" key)
+    event.preventDefault();
+
+    // Check if tag already exist
+    const newTagName = this.newTagTarget.value.trim();
+    if (newTagName.length == 0) {
+      return;
+    }
+
+    const tags = this.tagListTarget.value.split(",");
+    const index = tags.indexOf(newTagName);
+    if (index != -1) {
+      // highlight the value in red
+      this.newTagTarget.classList.add("tag-error");
+      return;
+    }
+
+    // add to tagList
+    this.tagListTarget.value = this.tagListTarget.value.concat(`,${newTagName}`);
+
+    // Create new li component with value
+    const newTagElement = this.templateTarget.content.cloneNode(true);
+    const spanElement = newTagElement.querySelector("span");
+    spanElement.innerText = newTagName;
+    this.listTarget.appendChild(newTagElement);
+
+    // Clear new tag value
+    this.newTagTarget.value = "";
+  }
+
+  removeTag(event) {
+    // Text to remove
+    const tagName = event.srcElement.previousElementSibling.textContent;
+
+    // Remove tag from list
+    const tags = this.tagListTarget.value.split(",");
+    const index = tags.indexOf(tagName);
+    tags.splice(index, 1);
+    this.tagListTarget.value = tags.join(",");
```

Comment:
What do you think of this one?

```suggestion
    // Remove tag from list
    const tags = this.tagListTarget.value.split(",");
    tags = tags.filter(tag => tag != tagName);
    this.tagListTarget.value = tags.join(",");
```

Or shorter:

```suggestion
    // Remove tag from list
    const tags = this.tagListTarget.value.split(",");
    this.tagListTarget.value = tags.filter(tag => tag != tagName).join(",");
```

Reply:
Nice one !